### PR TITLE
Add support for cross-entropy loss with integer labels

### PR DESCRIPTION
**Pull request by aliciafmachado**
Adds support for cross-entropy loss with integer labels (Issue #60) following Jax implementation in `softmax_cross_entropy_with_integer_labels` in the Optax library.

* Additionaly we clean-up one-hot encodings from task definition.
* We fix the training script so that it works with new loss.

### CODE REVIEW

File: animated-transformer/src/lib/gtensor/gtensor.ts [hunk 1]
```diff
@@ -799,39 +807,41 @@ export class GTensor<G extends DName> {
     return new GTensor<G>(this.tensor.equal(g2.tensor), this.dimNames);
   }
 
-  // TODO: support batched gather.
-  public gather<D extends G, G2 extends DName>(
+  public gather<D extends G, B extends G & G2, G2 extends DName>(
```
Comment (aliciafmachado) [inline animated-transformer/src/lib/gtensor/gtensor.ts:810]: Add a comment with a TODO and create a bug on tfjs to fix grad for a number of batch dimensions higher than one

File: animated-transformer/src/lib/transformer/common_transformer.ts [hunk 2]
```diff
@@ -153,6 +153,26 @@ export function allPastTokensCrossEntropyLoss(
     return crossEntropyLoss.tensor.asScalar();
 }
 
+/**
```
Comment (aliciafmachado) [inline animated-transformer/src/lib/transformer/common_transformer.ts:156]: Clean-up allPastTokensCrossEntropyLoss

**Review by iislucas — APPROVED** [review]

**Comment by iislucas** [comment]
If this is ready to merge, please do so, so I can unblock the big monster PR... https://github.com/PAIR-code/tiny-transformers/pull/63 :)
